File: app/src/main/java/com/shuyu/apprecycler/chat/data/factory/LocalChatDBFactory.java
package com.shuyu.apprecycler.chat.data.factory;

/**
 * 数据库工厂对象
 * Created by guoshuyu on 2017/9/11.
 */

public class LocalChatDBFactory {

    public static ILocalChatDBManager getChatDBManager() {
        //需要替换数据库框架，只需要替换这个实例化即可
        return LocalChatBDRealmManager.newInstance();
    }
}

File: app-kotlin/src/main/java/example/shuyu/recycler/kotlin/chat/data/factory/LocalChatDBFactory.kt
package example.shuyu.recycler.kotlin.chat.data.factory

import example.shuyu.recycler.kotlin.chat.data.factory.realm.LocalChatBDRealmManager
import example.shuyu.recycler.kotlin.chat.data.factory.room.LocalChatDBRoomManager

/**
 * 数据库工厂对象
 * Created by guoshuyu on 2017/9/11.
 */

public object LocalChatDBFactory {

    //需要替换数据库框架，只需要替换这个实例化即可
    val chatDBManager: ILocalChatDBManager
        //get() = LocalChatBDRealmManager.newInstance
        get() = LocalChatDBRoomManager.newInstance
}
